pandas/core/groupby/generic.py

```diff
@@ -1025,8 +1025,8 @@ def transform(self, func, *args, **kwargs):
             object.__setattr__(group, "name", name)
             res = wrapper(group)
 
-            if hasattr(res, "values"):
-                res = res.values
+            if hasattr(res, "_values"):
```

Comment:
could do

res = getattr(res, ‘_values’, res)

Reply:
I think Joris opened an issue about this recently; can we make this a more explicit isinstace ABCWhatever check?

Reply:
I think this is ref #27167 so isinstance seems to be the direction rather than getattr